Update README Based on Roadmap. Remove Stale Information (#151)
freeCodeCamp/chapter: README.md

* [Node.js](https://nodejs.org) / [Express](https://expressjs.com) for our backend using JavaScript/TypeScript
* [Postgres](https://www.postgresql.org) with [Sequelize ORM](https://sequelize.org)
* [Next.js](https://nextjs.org/) for both client and server-side rendering of the frontend (NextJS is based on [React](https://reactjs.org))
  * [JavaScript/TypeScript](https://www.typescriptlang.org/index.html#download-links)

>   * [Styled Components](https://www.styled-components.com) for styling.

Author: @vkWeb what is true, but in the last meeting we also decided on Javscript inside of Typescript files to not alienate Javascript contributors.  I'm happy to have this reworded, but it seemed pretty clear we were not 100% Typescript and someone else previously added this bullet point, so it's not really part of this PR.

### Installing Docker

> See the [Docker installation](https://docs.docker.com/v17.12/install/) "Supported Platforms" section and follow the instructions to download & install Docker Desktop for your operating system (or Docker CE for linux).

Reviewer: ```suggestion
See the [Docker installation](https://docs.docker.com/install/) "Supported platforms" section and follow the instructions to download & install Docker Desktop for your operating system (or Docker CE for linux).
```

Can be fixed in a separate PR, but the current Docker install link here (for v17.12) is out of date - this link ☝️ will always link to the latest release.

Author: Good suggestion.

The original change was to reword it to avoid "Click here", so if this is changed in another PR then my preference is to [avoid "Click here" links](https://www.w3.org/QA/Tips/noClickHere).

![a diagram illustrating the proposed schema for Chapter](data/schema.png)
</details>

## User Stories

### MVP
The [MVP user stories are shown in the MVP Project](https://github.com/freeCodeCamp/chapter/projects/1) kanban / cards and as [issues marked with "MVP"](https://github.com/freeCodeCamp/chapter/labels/MVP).

### Post-MVP

> We are maintaining a list of [post-MVP user stories](https://github.com/freeCodeCamp/chapter/issues/84).

Author: @vkWeb this is such a small concern, but from a technical writing standpoint we should never have a single bullet point. If/when a second bullet is added then we can certainly add bullets.

> We are maintaining a list of [post-MVP user stories](https://github.com/freeCodeCamp/chapter/issues/84).

Author: Sorry, I meant to say that the user stories were well out of sync with the other two MVP documents.  I don't see why we'd want to maintain the same information in 2 or 3 places unless someone is going to actively maintain all of the locations.

It's already enough work just to keep things up to date so my vote is for more links and less of a huge README that gets out of date every day.

> We are maintaining a list of [post-MVP user stories](https://github.com/freeCodeCamp/chapter/issues/84).

Author: Similar to the user stories and roadmap bullets, the Quick Links I added initially in the CONTRIBUTING.md are out of date.

My feeling is many of the bullets we initially added to the README and CONTRIBUTING made sense only temporarily because there was less structure and places to point newcomers in the initial chaos.

If/as other pages in the Github project go into more details it's fair to consider linking out to those sections to avoid huge and/or out of date .md files.

> We are maintaining a list of [post-MVP user stories](https://github.com/freeCodeCamp/chapter/issues/84).

Reviewer 2: @allella Umm.. Yes you are right. 

Let's keep it that way then ;)

### Suggesting a New Story

1. Please check your idea does not already exist as an [MVP issue](https://github.com/freeCodeCamp/chapter/labels/MVP), or in the [post-MVP list](https://github.com/freeCodeCamp/chapter/issues/84).

> 1. Create a new "Issue" and post details about the suggested user story.

Author: I'd be glad to move this to CONTRIBUTING.md if other find it helpful, but I'm not going to maintain that in needs to be in the README.

> 1. Create a new "Issue" and post details about the suggested user story.

Author: We do have a similar reminder in the CONTRIBUTING.md.  So, I'd suggest we update the CONTRIBUTING.md to link to the MVP project and #84 as a quick view of what's already in the works or been discussed.

> 1. Create a new "Issue" and post details about the suggested user story.

Reviewer 2: @allella Okay fine, we can keep it in contributing.md.

## Roadmap

> The on-going [project Roadmap conversation](https://github.com/freeCodeCamp/chapter/issues/47) is regularly updated to reflect the overall progress and for higher-level discussions.

Reviewer 2: Add this below the old roadmap. Something like:

```suggestion
1. Design the schema - Done.
2. Set up the API endpoints - Done.
3. Built the web client and let other developers use the API to build mobile clients and voice interface clients.

We have a more detailed information on roadmap on the [Roadmap issue]https://github.com/freeCodeCamp/chapter/issues/47). It is regularly updated to reflect the overall progress and for higher-level discussions.
```

Author: If someone else is going to maintain the bullets here then I'm fine with it. As it was, this list was out of sync with #47, which has the established roadmap and is much easier / faster to edit.

Reviewer 2: @allella Link to the issue seems right.

Quincy Larson is the project lead. [FreeCodeCamp](https://www.freecodecamp.org) will start "dogfooding" the MVP with several of its local study groups.

Here's an out-dated example of an app with similar functionality: [The freeCodeCamp Study Group Directory](https://study-group-directory.freecodecamp.org).

## Contributing

> * You should [join our Discord server](https://discord.gg/vbRUYWS) to get connected with people interested in this project and to be aware of our future announcements.

Reviewer 2: Not required.

Author: This was previously in the README and I moved it. If we want to move it to CONTRIBUTING.md then that's fine with me.

Author: Discord is similarly mentioned in the CONTRIBUTING.md.

This note already existed in the README, so I have no strong feelings that it needs to be in the README.

However, if we leave it then I'd vote to move it out of the Roadmap section and into the Contributing section.

Reviewer 2: @allella There's no harm in keeping it in readme's roadmap and contributing. Let it be.

> * Please read the [**suggested steps to contribute code to the Chapter project**](CONTRIBUTING.md) below forking or submitting any pull requests.

Reviewer: ```suggestion
* Please read the [**suggested steps to contribute code to the Chapter project**](CONTRIBUTING.md) before forking or submitting any pull requests.
```

Author: Thanks Scott for the typo fix. @vkWeb had suggested not having this new bullet.  I don't think it's critical and perhaps it's better in the CONTRIBUTING.md.

The purpose was to more to encourage new user story suggestions to create a new issue instead of dropping new ideas in comments, where it's likely to get lose.

The bullet about checking that the idea doesn't already exist was secondary, but still a reminder for anybody who might not know where to look for the full set of existing suggestions.

## License

Copyright © 2019 freeCodeCamp.org

> The computer software is licensed under the [BSD-3-Clause](LICENSE) license.

Reviewer 2: Revert to old one.

Author: We should avoid single bullet points, so I'm going to suggest we remove the bullet.
